lib/grpcServer/handlers/getTransactionsByFilterHandlerFactory.js

```diff
@@ -1,12 +1,52 @@
-const { RawTransaction } = require('@dashevo/dapi-grpc');
+const BloomFilter = require('bloom-filter');
+const { RawTransaction, TransactionFilterResponse, RawMerkleBlock } = require('@dashevo/dapi-grpc');
 
-function getTransactionsByFilterHandlerFactory() {
-  function getTransactionsByFilterHandler(call, callback) {
-    const transaction = new RawTransaction();
+const BloomFilterEmitter = require('../../bloomFilter/BloomFilterEmitter');
 
-    transaction.setBytes(Buffer.from('000000000000000', 'hex'));
+/**
+ *
+ * @param {BloomFilterCollection} bloomFilterCollection
+ * @param {testTransactionAgainstFilter} testTransactionAgainstFilter
+ * @return {getTransactionsByFilterHandler}
+ */
+function getTransactionsByFilterHandlerFactory(
+  bloomFilterCollection,
+  testTransactionAgainstFilter,
+) {
+  function getTransactionsByFilterHandler(call) {
+    const filter = new BloomFilter(call.request);
 
-    callback(null, transaction);
+    const filterEmitter = new BloomFilterEmitter(filter, testTransactionAgainstFilter);
+
+    filterEmitter.on('contains', (transaction) => {
+      const rawTransaction = new RawTransaction();
+
+      rawTransaction.setIsInstantSend(transaction.instantlock);
+      rawTransaction.setData(transaction.serialize());
+
+      const response = new TransactionFilterResponse();
+      response.setTransaction(rawTransaction);
+
+      call.write(response.toObject());
+    });
+
+    filterEmitter.on('block', (merkleBlock) => {
+      const rawMerkleBlock = new RawMerkleBlock();
+      rawMerkleBlock.setData(merkleBlock.serialize());
+
+      const response = new TransactionFilterResponse();
+      response.setMerkleBlock(rawMerkleBlock);
+
+      call.write(response.toObject());
+    });
+
+    bloomFilterCollection.add(filterEmitter);
+
+    call.on('cancelled', () => {
+      call.end();
+
+      bloomFilterCollection.add(filterEmitter);
```

Comment:
Not sure I get that part. Why do we want to add bloom filter to the collection on cancelled call?